### DISCUSSION TITLE
SortableHandle as in react-sortable-hoc

### DISCUSSION THREAD

**Discussion by krau5** [clauderic/dnd-kit#603] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/52201020/150983322-61eb5a68-15c9-49ab-9b42-7cb704e8570f.png)
How to make a handle to drag a whole component using `useSortable`?

**Comment by clauderic** [comment, marked as answer]
You can refer to the documentation on this here:

https://docs.dndkit.com/api-documentation/draggable#flexibility


For `useSortable`, it's nearly the same idea as with `useDraggable`, all you need to do is attach the listeners and attributes to the drag handle element rather than the wrapper element that receives the `ref` and `transform` styles:

```ts
import React from 'react';
import {useSortable} from '@dnd-kit/sortable';
import {CSS} from '@dnd-kit/utilities';

function SortableItem(props) {
  const {
    attributes,
    listeners,
    setNodeRef,
    transform,
    transition,
  } = useSortable({id: props.id});
  const style = {
    transform: CSS.Transform.toString(transform),
    transition,
  };

const dragHandle = <button {...listeners} {...attributes}>::</button>;
  
  return (
    <li ref={setNodeRef} style={style}>
      {dragHandle}
      Card title
    </li>
  );
}
```

**Reply by krau5** [reply]
```
export const DraggableComponent: React.FunctionComponent<Props> = ({ id }) => {
  const { transition, transform, attributes, listeners, setNodeRef } = useSortable({ id });

  const style = {
    transform: CSS.Transform.toString(transform),
    transition,
  };

  return (
    <div css={styles.card} style={style} ref={setNodeRef}>
      <span css={styles.trigger} {...listeners} {...attributes}>
        <Icon name="drag_indicator" />
      </span>

      <Typography>Hello</Typography>
    </div>
  );
};
```

I'm terrible sorry, but `transform` is `null`. I have even made it without a separated drag handle.
What should I do?

**Reply by clauderic** [reply]
What you have above looks okay, nothing stands out.

Open a CodeSandbox replicating the issue if you need further assistance. I can't help debug just seeing isolated snippets of code unfortunately, as the issue could be elsewhere in your code.
